src/gui/qgsmaptoolidentify.cpp

```diff
@@ -258,23 +256,23 @@ bool QgsMapToolIdentify::identifyVectorLayer( QList<IdentifyResult> *results, Qg
     filter = renderer->capabilities() & QgsFeatureRenderer::Filter;
   }
 
-  for ( ; f_it != featureList.end(); ++f_it )
+  for ( const QgsFeature &feature : qgis::as_const( featureList ) )
   {
     QMap< QString, QString > derivedAttributes = commonDerivedAttributes;
 
-    QgsFeatureId fid = f_it->id();
-    context.expressionContext().setFeature( *f_it );
+    QgsFeatureId fid = feature.id();
+    context.expressionContext().setFeature( feature );
 
-    if ( filter && !renderer->willRenderFeature( *f_it, context ) )
+    if ( filter && !renderer->willRenderFeature( const_cast<QgsFeature &>( feature ), context ) )
```

Comment:
Does anyone have an idea why `willRenderFeature` needs a non const feature? Should we rather send it over a copy instead of casting the constness away?

Reply:
Only historical reasons I think - just like renderFeature() and symbolForFeature(). We could probably fix that without breaking python API in 3.x. Passing a copy seems like an overkill...